airbyte-integrations/connectors/source-iterable/unit_tests/test_extractors.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+from io import StringIO
+from unittest.mock import Mock
+
+import pytest
+from source_iterable.components import (  # Import your module containing the classes
+    EventsRecordExtractor,
+    ListUsersRecordExtractor,
+    XJsonRecordExtractor,
+)
+
+
+@pytest.fixture
+def mock_response():
+    mock_response = Mock()
+    return mock_response
+
+@pytest.mark.parametrize("extractor_class,field_path", [(ListUsersRecordExtractor, "getUsers"), (XJsonRecordExtractor, "users"), (EventsRecordExtractor, "events")])
+def test_extract_records(mock_response, extractor_class, field_path):
```

Comment:
Nit: can you split these into 3 separate tests?

Reply:
Updated

Reply:
Thank you!